fahrschein/src/main/java/org/zalando/fahrschein/CursorOffsetCommitException.java

```diff
@@ -0,0 +1,58 @@
+package org.zalando.fahrschein;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.zalando.fahrschein.domain.Cursor;
+
+@SuppressWarnings("serial")
+public class CursorOffsetCommitException extends IOException {
+    private final int statusCode;
+    private final Cursor cursor;
+    private final String subscriptionId;
+
+    public CursorOffsetCommitException(int statusCode, Cursor cursor, String subscriptionId, IOException cause) {
+        super(formatMessage(statusCode, cursor, subscriptionId), cause);
+        this.statusCode = statusCode;
+        this.cursor = cursor;
+        this.subscriptionId = subscriptionId;
+    }
+
+    public CursorOffsetCommitException(int statusCode, Cursor cursor, String subscriptionId) {
+        super(formatMessage(statusCode, cursor, subscriptionId));
+        this.statusCode = statusCode;
+        this.cursor = cursor;
+        this.subscriptionId = subscriptionId;
+    }
+
+    private static String formatMessage(int statusCode, Cursor cursor, String subscriptionId) {
+        String msg;
+        switch (statusCode) {
+            case 422:
+                msg = String.format("Cursor for subscription [%s] to event [%s] in partition [%s] with offset [%s] failed to commit because of error 422 (Unprocessable Entity). This likely means that the processing time of the batch exceeded the timeout of 60 seconds. In such case, you may want to investigate the slowness in the processing, and/or reduce the batch size", 
```

Comment:
Could you break that line so no horizontal scrolling is needed in Github?

Reply:
(In some cases, the user might have set the commit timeout smaller than 60 seconds, so I'm not sure this needs to be mentioned in the text.)

Reply:
adjusted